expo/expo
# [docs] Configure and integrate Vale in docs

# Why



Closes ENG-9430 ENG-9431 ENG-9985 ENG-9432

# How



- By initializing vale in the `docs` directory and then adding rules from our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md). 
- The initial configuration file can be found in `docs/.vale.ini`
- Rules can be found in `.vale/writing-styles/`
- Currently, this PR adds custom rules ~~and rules already defined by ["write-good"](https://github.com/errata-ai/write-good) guide.~~ We are not integrating write-good rules in our system right now. We'll be only relying on our Custom rules from the writing style guide.
- All configuration and custom rules are set to the "warning" level.
- Also added instructions on how to set up Vale in `docs/README.md`.

# Test Plan

- Go to docs repo
- Run `yarn` to install vale related package (which is required to run the script "lint-prose" if you don't have vale installed locally
- Run `yarn run lint-prose`

Alternatively, you can also
- Install [globally on your system](https://vale.sh/docs/vale-cli/installation/). For example, with homebrew: `brew install vale`
- Then, [install VSCode extension](https://marketplace.visualstudio.com/items?itemName=errata-ai.vale-server).
- This will eliminate the need to run the "lint-prose" script and you can see the warnings in VSCode editor.



# Checklist
